lib/Telemetry/Monitor.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ConfigReport\Telemetry;
+
+use OC\License\LicenseFetcher;
+use OCA\ConfigReport\ReportDataCollector;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\ILogger;
+
+class Monitor {
+	private IConfig $config;
+	private ILogger $logger;
+	private IClientService $clientService;
+	private ReportDataCollector $collector;
+
+	public function __construct(IConfig $config, ILogger $logger, IClientService $clientService, ReportDataCollector $collector) {
+		$this->config = $config;
+		$this->logger = $logger;
+		$this->clientService = $clientService;
+		$this->collector = $collector;
+	}
+
+	public function sendTelemetry(): void {
+		$telemetryOptIn = $this->config->getSystemValue('telemetry.enabled', true);
```

Comment:
It might be better to default it to false, or maybe depending on the license.
If the default is true, I'd consider it a change in the behavior, which looks bad (plus privacy concerns when the app hasn't sent data anywhere until now).
Also, I'm pretty sure people won't read the release notes for just an upgrade.

Reply:
We discussed this. There is no GDPR issue here as there is no sensitive data sent. Adding configurable additional email targets as suggested above gives even more relaxation. You could use the statement reading release notes in both directions. True is better as we then know the installed base.

Reply:
At least in the example in the description sensitive data is sent:
1. `"server OS version": "Linux deepdiver` -> The second is the (internal) hostname of the target system
2. `"overwrite.cli.url": "http://localhost",` -> Could contain a sensitive (internal) domain name
3. ` "datadirectory"` -> Can contain sensitive (internal) path names
4. `apps_paths`  -> Can contain sensitive (internal) path names

Besides that this should be really default to `false` instead of `true` :-1: , you shouldn't do any opt-out by default and choose the privacy friendly opt-in instead.

Reply:
We have internal requirements which state that telemetry is opt-out. 🤷

Reply:
Then show a pop-up similar to the first run wizard to the admin after the update with a simple opt-out possibility (e.g. licking on "no, thanks i don't want to have my data sent to some external system).

With this feature included and enabled by default you will most likely loose the last community user to e.g. Nextcloud and other alternatives.

Reply:
Community is not affected. Customers will be taken care of in different communication ways.
Thx